src/components/search/data/hooks.js

```diff
@@ -126,18 +86,18 @@ export const getCatalogString = (catalogs) => {
 
 export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan, offerCatalogs = [] }) => {
   // default to showing all catalogs
-  const { showAllCatalogs, setShowAllCatalogs } = useContext(SearchContext);
+  const { refinementsFromQueryParams, dispatch } = useContext(SearchContext);
 
   useEffect(() => {
     // if there are no subscriptions or offers, we default to showing all catalogs
     if (!subscriptionPlan && offerCatalogs.length < 1) {
-      setShowAllCatalogs(true);
+      dispatch(setRefinementAction(SHOW_ALL_NAME, 1));
     }
   }, [subscriptionPlan, offerCatalogs.length]);
 
   const filters = useMemo(
     () => {
-      if (showAllCatalogs) {
+      if (refinementsFromQueryParams[SHOW_ALL_NAME]) {
         // show all enterprise catalogs
         return `enterprise_customer_uuids:${enterpriseConfig.uuid}`;
       }
```

Comment:
While we're in this file... 

Now that the API request to fetch offers (codes) is behind the feature flag, we should be safe to remove `features.ENROLL_WITH_CODES` from the conditionals on line 107 and 113. Without making that API request, `offerCatalogs` should always be an empty array based on the initial offers reducer state.

Reply:
I'm a bit wary of that, so I'm gonna save it for the next PR. :)